Sources/Moya/Response.swift

```diff
@@ -127,10 +127,26 @@ public extension Response {
             guard let jsonObject = (try mapJSON() as? NSDictionary)?.value(forKeyPath: keyPath) else {
                 throw MoyaError.jsonMapping(self)
             }
-            do {
-                jsonData = try JSONSerialization.data(withJSONObject: jsonObject)
-            } catch {
-                throw MoyaError.jsonMapping(self)
+            if JSONSerialization.isValidJSONObject(jsonObject) {
+                do {
+                    jsonData = try JSONSerialization.data(withJSONObject: jsonObject)
+                } catch {
+                    throw MoyaError.jsonMapping(self)
+                }
+            } else {
+                let wrappedJsonObject = ["value": jsonObject]
+                let wrappedJsonData: Data
+                if JSONSerialization.isValidJSONObject(wrappedJsonObject) {
```

Comment:
It looks like we're repeating this block again. Can we extract it into a closure?